gammapy/datasets/make.py

```diff
@@ -356,6 +364,213 @@ def make_test_bg_cube_model(detx_range=Angle([-10., 10.], 'degree'),
         mask = (detx_points <= detx_center) & (dety_points <= dety_center)
         background = background*mask
 
-    bg_cube_model.background = Quantity(background, '1 / (s TeV sr)')
+    bg_cube_model.data = Quantity(background, '1 / (s TeV sr)')
 
     return bg_cube_model
+
+
+def make_test_dataset(fits_path, overwrite=False,
+                      observatory_name='HESS', n_obs=10,
+                      datestart=None, dateend=None,
+                      random_state='random-seed'):
+    """
+    Make a test dataset and save it to disk.
+
+    Uses:
+
+    * `~gammapy.datasets.make_test_observation_table` to generate an
+      observation table
+
+    * `~gammapy.datasets.make_test_evenlist` to generate an event list
+      and effective area table for each observation
+
+    * `~gammapy.obs.DataStore` to handle the file naming scheme;
+      currently only the H.E.S.S. naming scheme is supported
+
+    See also :ref:`background_make_models_datasets_for_testing`.
+
+    Parameters
+    ----------
+    fits_path : str
+        Path to store the files.
+    overwrite : bool, optional
+        Flag to remove previous datasets in fits_path (if existing).
+    observatory_name : str, optional
+        Name of the observatory; a list of choices is given in
+        `~gammapy.obs.observatory_locations`.
+    n_obs : int
+        Number of observations for the obs table.
+    datestart : `~astropy.time.Time`, optional
+        Starting date for random generation of observation start time.
+    dateend : `~astropy.time.Time`, optional
+        Ending date for random generation of observation start time.
+    random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}, optional
+        Defines random number generator initialisation.
+        Passed to `~gammapy.utils.random.get_random_state`.
+    """
+    # create output folder
+    outdir = fits_path
+    if not os.path.isdir(outdir):
```

Comment:
This code shows up twice in this PR.
I haven't thought about if it's a good idea to do this, but if you think it is, OK to keep as is, but please split this out into a private utility function and call it from both places.